src/charm.py

```diff
@@ -59,8 +59,9 @@ def _mongodb_relation_joined(self, event) -> None:
             )
             if not mongo_peer.is_ready(standalone=True):
                 logger.debug(
-                    "unit: %s is not ready, cannot initilise replica set until all units are ready, deferring on relation-joined",
-                    peer,
+                    "unit: {} is not ready, cannot initilise replica set until all units are ready, deferring on relation-joined".format(
+                        peer
+                    )
```

Comment:
Logging is the one place where you should use f-strings or `.format` - please revert changes like these in this PR. The reasoning here is not to preemptively "build strings" that may not need to be evaluated depending on the log level the program is run with.

Using `.format` anywhere outside of `logger.foo()` is fine :)